### PR TITLE
Improvements to UDP commands

### PR DESCRIPTION
**Pull request by yuguorui**
Hi~ I made some improvements to the UDP command, the following is a summary:
1. Use u8:MAX instead of the deprecated u8::max_value().
2. Add a flag to toggle UDPAssociate support in the server side.
3. Add a new unit test for UDPAssociate.
4. Update the README.

Any suggestions for improvement are welcome.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -346,45 +353,97 @@ mod test {
...
-        let mut buf = [0; 13];
-        let (len, addr) = socket.recv_from(&mut buf).await?;
-        assert_eq!(len, 12);
-        assert_eq!(&buf[..12], b"hello world!");
+            let (tx, rx) = oneshot::channel();
+            tokio::spawn(setup_socks_server("[::1]:0", None, tx));
+            let backing_socket = TcpStream::connect(rx.await.unwrap()).await.unwrap();
 
-        socket.send_to(b"hello world!", addr).await?;
+            let tunnel = client::Socks5Datagram::bind(backing_socket, "[::]:0")
```
Comment (dizda) [inline src/lib.rs:406]: more comment is required here 🙏

**Comment by dizda** [comment]
Just tested it, that looks brillant @yuguorui !

File: src/lib.rs (hunk 1 above)
Comment (yuguorui) [reply src/lib.rs:410]: Sure!

I've added additional comments here, and also added additional documentation and examples at the Socks5Datagram::bind definition.

**Comment by dizda** [comment]
Brillant, thank you @yuguorui !

**Pull request merged**